tests/src/test/scala/org/apache/openwhisk/core/limits/ActionLimitsTests.scala

```diff
@@ -463,4 +463,33 @@ class ActionLimitsTests extends TestHelpers with WskTestHelpers with WskActorSys
       _.response.result.get.fields("error") shouldBe Messages.memoryExhausted.toJson
     }
   }
+
+  /**
+   * Test that a heavy logging action is interrupted within its timeout limits.
+   */
+  it should "interrupt the heavy logging action within its time limits" in withAssetCleaner(wskprops) {
+    (wp, assetHelper) =>
+      val name = s"PythonTestLoggingActionCausingTimeout-${System.currentTimeMillis()}"
+      print(s"\n create action ${name} using api host: ${wskprops.apihost}..")
+      assetHelper.withCleaner(wsk.action, name, confirmDelete = true) { (action, _) =>
+        action.create(
+          name,
+          Some(TestUtils.getTestActionFilename("loggingTimeout.py")),
+          timeout = Some(allowedActionDuration))
+      }
+      val durationMillis = allowedActionDuration + 3.minutes
+      val checkDurationMillis = allowedActionDuration + 1.minutes
+      val run =
+        wsk.action.invoke(name, Map("durationMillis" -> durationMillis.toMillis.toJson, "delayMillis" -> 100.toJson))
+      withActivation(wsk.activation, run) { result =>
+        withClue("Activation result not as expected:") {
+          result.response.status shouldBe ActivationResponse.messageForCode(ActivationResponse.DeveloperError)
+          result.response.result.get.toString should include("exceeded its time limits")
+          result.logs.get.last should include(Messages.logFailure)
+          val startLogMillis = Instant.parse(result.logs.get.head.split(' ')(0)).toEpochMilli()
+          val endLogMillis = Instant.parse(result.logs.get.last.split(' ')(0)).toEpochMilli()
+          (endLogMillis - startLogMillis).toInt should be < (checkDurationMillis).toMillis.toInt
```

Comment:
There's quite a bit of deduplication that can be done here. `result.logs.get` is used a couple of times.

What about:

```scala
val logs = result.logs.get
logs.last should include(Messages.logFailure)

val parseLogTime = line => Instant.parse(line.split(' ').head)
val startTime = parseLogTime(logs.head)
val endTime = parseLogTime(logs.last)

Duration.between(startTime, endTime) should be < (checkDuration)
```

You'll need to import [TimeHelpers.scala](https://github.com/apache/incubator-openwhisk/blob/be1e3a19c02956c9be85023a0bb0ff399c21444d/common/scala/src/main/scala/org/apache/openwhisk/utils/TimeHelpers.scala) to implicitly transform the java.time.Duration from Duration.between to a Scala duration.

Reply:
Applied.